### PR TITLE
GenAI Reorg Part 5: Create PromptExample

### PR DESCRIPTION
**Pull request by brendanshean**
## WHAT
Create `Evidence::Research::GenAI::TestExample` model

## WHY
[Prompt examples](https://www.notion.so/quill/Internal-Tool-Organizing-Trials-into-Datasets-8dbc188caf194157a2a28395e907ddc0?pvs=4#78a75b19fb6046808946eaebe9d97f5c) are components of `Dataset` used to provide n-shot in our `LLMPrompt`
  
## HOW
Write a migration and a model file.

NB:  There is no validation for locked on `Dataset` since we want to allow for the addition of `PromptExample` to a dataset. 

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Internal-Tool-Organizing-Trials-into-Datasets-8dbc188caf194157a2a28395e907ddc0?pvs=4

### What have you done to QA this feature?
(Provide enough detail that a reviewer could assess whether additional QA should be done. For larger projects, additionally use the Engineer Feature Testing Notion template. Review Guidelines if needed: https://www.notion.so/quill/Github-PR-QA-Guidelines-49e99fc965654ceeb8c6249bd9d181d7)

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  YES
Have you deployed to Staging? | YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES

### CODE REVIEW

File: services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/prompt_example.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    module GenAI
+      class PromptExample < ApplicationRecord
+        HUMAN_STATUSES = [
+          OPTIMAL = 'optimal',
+          SUB_OPTIMAL = 'sub_optimal'
+        ].freeze
+
+        belongs_to :dataset
+
+        validates :human_status, presence: true
```
Comment (dandrabik) [inline services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/prompt_example.rb]: Likely not worth re-opening a discussion, but this is still a little weird for a name.  Sounds like it's storing the status of a human (alive? tired? hungry?). We might consider something like `human_assigned_status` or `staff_assigned_status`

File: services/QuillLMS/engines/evidence/db/migrate/20240619224707_create_evidence_research_gen_ai_prompt_examples.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class CreateEvidenceResearchGenAIPromptExamples < ActiveRecord::Migration[7.0]
+  def change
+    create_table :evidence_research_gen_ai_prompt_examples do |t|
+      t.integer :dataset_id, null: false
+      t.text :student_response, null: false
+      t.string :human_status, null: false
+      t.text :human_feedback
```
Comment (dandrabik) [inline services/QuillLMS/engines/evidence/db/migrate/20240619224707_create_evidence_research_gen_ai_prompt_examples.rb]: Again, not to reopen it, but something like `staff_feedback` is more specific and less weird.

**Review by dandrabik — APPROVED** [review]
LGTM. I left comments on the `human_` names, but you can take or leave based on your appetite to discuss naming anymore (might just be better to move on).

File: services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/prompt_example.rb (hunk 1 above)
Comment (brendanshean) [reply services/QuillLMS/engines/evidence/app/models/evidence/research/gen_ai/prompt_example.rb]: That's a good point.  I will update these in the next PR.

**Pull request merged**